quinn-proto/src/cid_generator.rs

```diff
@@ -0,0 +1,54 @@
+use rand::RngCore;
+
+use crate::shared::ConnectionId;
+use crate::MAX_CID_SIZE;
+
+/// Generates connection IDs for incoming connections
+pub trait ConnectionIdGenerator: Send {
+    /// Generates a new CID
+    fn generate_cid(&mut self) -> ConnectionId;
```

Comment:
We should have a brief note here describing desirable properties for connection IDs, e.g. that they be indistinguishable from random data by an external viewer.

Reply:
I will add some notes from RFC doc on Connection ID properties